hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -2115,14 +2118,27 @@ func (r *NodePoolReconciler) getConfig(ctx context.Context,
 	configs = append(configs, nodeTuningGeneratedConfigs.Items...)
 
 	for _, config := range configs {
-		manifestRaw := config.Data[TokenSecretConfigKey]
-		manifest, err := defaultAndValidateConfigManifest([]byte(manifestRaw))
-		if err != nil {
-			errors = append(errors, fmt.Errorf("configmap %q failed validation: %w", config.Name, err))
-			continue
+		cmPayload := config.Data[TokenSecretConfigKey]
+		// ignition config-map payload may contain multiple manifests
+		yamlReader := yaml.NewYAMLReader(bufio.NewReader(strings.NewReader(cmPayload)))
```

Comment:
For any other reviewer wondering why we need `bufio` on a string: it's because it's explicit in [the function signature](https://github.com/kubernetes/kubernetes/blob/790dfdbe386e4a115f41d38058c127d2dd0e6f44/staging/src/k8s.io/apimachinery/pkg/util/yaml/decoder.go#L303-L304).